packages/utils/src/cache/messenger.ts

```diff
@@ -0,0 +1,122 @@
+import { MESSENGER_PREFIX } from './constants.js';
+import { Logger } from './helpers.js';
+
+export type Message = {
+  id: number;
+  key: string;
+  value: string;
+}
+
+export type OnMessageCallback = (message: Message) => void;
+
+export type OnCompleteCallback = () => void;
+
+enum StorageType {
+  MessagePosts='messagePosts',
+}
+
+type StorageNames = {
+  [name in StorageType]: string
+};
+
+/**
+ * Cache messenger manage post/receive to others cache messenger
+ */
+export class CacheMessenger {
+
+  /**
+   * Channel to send message
+   */
+  protected broadcastChannel: BroadcastChannel;
+
+  /**
+   * Total messages post state, shared across others messenger
+   */
+  protected totalPost = 0;
+
+  /**
+   * Names for manage all states temporary
+   */
+  private storageNames: StorageNames;
+
+  /**
+   * Messaging callback
+   */
+  public onMessage: OnMessageCallback | undefined = undefined;
+
+  /**
+   * Constructor
+   * @param name messenger name
+   */
+  constructor (name: string) {
+    const messengerName = `[${MESSENGER_PREFIX}][${name}]`;
+    this.broadcastChannel = new BroadcastChannel(messengerName);
+    this.storageNames = {
+      messagePosts: `${messengerName}[message-posts]`
```

Comment:
Can we utilise TypeScript's template literal here ?

Reply:
Refactored